### ISSUE TITLE
widgets module not ported (yet?)

### ISSUE THREAD

**Issue by nicolasmartin3d** [msorvig/qt5-qtbase-nacl#3]
Hi there,

I've been playing a little with this repo and my initial hope was that I could get some examples using the widgets module to work.

I followed your example in the declarative repo, and fixed a widget example using Q_GUI_MAIN.
Under Linux, I could just create a QApplication and some examples from the widgets directory (calculator for example) were kind of working (with some quirks, but overall there was hope!).
Under MacOsX, this did not work as it complains that a QCoreApplication already exists, being a QGuiApplication instantiated in qGuiStartup.

I'm not sure why I did not get the same error under Linux, but that's not the point :) Your documentation does not say if widgets are useable, but I guess they are not. Am I right ?

If that's the case, I would gladly try to help porting the widgets module. It seems I should add a pp::Module and pp::Instance class specific to widgets, that would use a QApplication, and some kind of Q_WIDGET_MAIN wrapper ? Has this module not been ported yet, because it really is way more complicated than that ?
Thanks !

**Comment by msorvig** [comment]
Hi, you are indeed correct in that a Q_WIDGET_MAIN wrapper is missing. I went ahead and implemented it now; there should be a widgets example in tests/manual/nacl/window_widgets .

There seems to be several DPI/font size, styling and layout issues (unless I made a mistake in my hurried example) - that could be something to investigate for those who are interested in getting widgets working.

I can take "trivial" patches here (we have a limit of 10 lines of code), larger patches need to go though the standard Qt contribution system at qt.io, once we get that set up for this project.
